main.cpp

```diff
@@ -20,67 +20,513 @@ You are going to write 10 UDTs in project3.
+    int numVacuumCleaners = 3; //                   3) member variables with relevant data types.
+    //number of eco-friendly cleaning supplies      
+    int numEcoFriendlyCleaningSupplies = 20;     
+    //stores the amount of water used per week.     
+    float waterUsedPerWeek = 200.f;            
+    //stores amount of profit made per week         
+    float profitPerWeek = 495.95f;               
+    //number of cars serviced per day               
+    int numberOfCarsServiced = 10;               
+    
+    struct Car //5)                                 Note that the nested type 'Car' is related to the 'CarWash' 
+    {
+        //2) member variables with relevant data types.  the names are appropriate for the U.D.T.'s purpose.
+        bool isAPickupTruck = false;
+        float gasMileage = 26.2f;        
+        int year = 1985;
+        std::string manufacturer = "Toyota";
+        std::string model = "Corolla";
+
+        //3) a member function whose parameter has a default value.
+        //the parameter name is related to the work the function will perform.
+        void fillTank(double fuelAmountInGallons = 2.0);  
+        void breakDown(std::string failureType, bool requiresTow = false);
+        int getMilesTraveledAnnually(bool includeUberLyftTrips);
+    };
+
+    //wash and wax car
+    void washAndWaxCar( Car car ); //4) a member function whose parameter is a UDT.
+    //charge customer
+    float chargeCustomer(float discountPercentage);
+    //detail the car interior
+    void detailInterior( Car car );
+    
+    //5) a member variable whose type is a UDT.
+    Car carBeingServiced;  
+};
+
+
+/*
+Thing 1) shoe
+5 properties:
+    1) brand 
+    2) size
+    3) color
+    4) condition
+    5) securing method
+3 things it can do:
+    1) be put on
+    2) be taken off
+    3) be thrown away
+ */
+
+struct shoe
+{
+
+    //brand
+    std::string brand = "Nike";
+    //size
+    double size = 12.5;
+    //color
+    std::string color = "white";
+    //condition
+    std::string condition = "good";
+    //securing method
+    std::string securingMethod = "velcro";
+
+    //be put on
+    void putOn ( shoe shoe );
+    //be taken off
+    void takeOff ( shoe shoe );
+    //be thrown away
+    void throwAway ( shoe shoe );
+
+};
+
+/*
+Thing 2) Wurlitzer
+5 properties:
+    1) volume level
+    2) vibrato amount
+    3) model
+    4) condition
+    5) number of keys
+3 things it can do:
+    1) change volume 
+    2) be played
+    3) turn on or off
+ */
+
+struct wurlitzer
+{
+
+    //volume level
+    int volumeLevel = 6;
+    //vibrato amount
+    double vibratoAmount = .5;
+    //model
+    std::string model = "200a";
+    //condition
+    std::string condition = "good";
+    //number of keys
+    int numKeys = 49;
+
+    //change volume
+    int changeVolume ( int knobPosition );
+    //be played
+    void play ( wurlitzer wurlitzer );
```

Comment:
how about 'play note', and the function arguments are details about that note, like note number, velocity, etc..
- [ ] revise